File: codility/src/main/java/codility/com/lessons/l02arrays/CyclicRotation.java
package codility.com.lessons.l02arrays;

/**
 * https://app.codility.com/programmers/lessons/2-arrays/cyclic_rotation/
 * 100%
 * @author Rodolfo Pereira Araujo
 */
public class CyclicRotation {
	public int[] solution(int[] A, int K) {
		if (K == 0 || A.length == 0) {
			return A;
		}

		K = K % A.length;
		if (K == 0) {
			return A;
		}

		int resposta[] = new int[A.length];
		for (int i = 0; i < A.length; i++) {
			int index = i + K;
			index = index >= A.length ? index - A.length : index;
			resposta[index] = A[i];
		}
		return resposta;
	}

}

File: codility_new/cyclic_rotation.py
# https://app.codility.com/programmers/lessons/2-arrays/cyclic_rotation/
def shift_right(A):
    if len(A) == 0:
        return
    previous = A[0]
    for i in range(1, len(A)):
        temp = A[i]
        A[i] = previous
        previous = temp
    A[0] = previous


def solution(A, K):
    if len(A) == 0 or K == len(A):
        return A
    K = K % len(A)
    for _ in range(K):
        shift_right(A)
    return A
  
